tests/cases/conformance/jsdoc/jsdocStringLiteral.ts

```diff
@@ -0,0 +1,12 @@
+// @allowJs: true
+// @filename: in.js
+// @out: out.js
+/**
+ * @param {'literal'} p1
+ * @param {"literal"} p2
+ * @param {'literal' | 'other'} p3
+ * @param {'literal' | number} p4
+ */
+function f(p1, p2, p3, p4) {
+    return p1 + p2 + p3 + p4 + '.';
```

Comment:
Good idea to add a string-literal-specific test. I didn't even know that we offered completions in strings. 

This test checks that the jsdoc parser gets the types right. It's in a JavaScript file so the parameter types would otherwise be any. That will produce a lot of observable differences in parameter help and quick info. But I trust those work if the checker can get the types right.